src/prefect/_internal/concurrency/supervisors.py

```diff
@@ -145,12 +153,21 @@ class Supervisor(abc.ABC, Generic[T]):
     Work sent to the supervisor will be executed when the owner waits for the result.
     """
 
-    def __init__(self, submit_fn: Callable[..., concurrent.futures.Future]) -> None:
+    def __init__(
+        self,
+        submit_fn: Callable[..., concurrent.futures.Future],
+        timeout: Optional[float] = None,
```

Comment:
I'm going to need to find a clean way to expose this in the `from_sync` and `from_async` interfaces.